modules/openxr/extensions/openxr_fb_space_warp_extension_wrapper.cpp

```diff
@@ -0,0 +1,190 @@
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "openxr_fb_space_warp_extension_wrapper.h"
+
+#include "../openxr_interface.h"
+#include "platform_gl.h"
+
+OpenXRFbSpaceWarpExtensionWrapper *OpenXRFbSpaceWarpExtensionWrapper::singleton = nullptr;
+
+OpenXRFbSpaceWarpExtensionWrapper *OpenXRFbSpaceWarpExtensionWrapper::get_singleton() {
+	return singleton;
+}
+
+OpenXRFbSpaceWarpExtensionWrapper::OpenXRFbSpaceWarpExtensionWrapper() {
+	singleton = this;
+}
+
+OpenXRFbSpaceWarpExtensionWrapper::~OpenXRFbSpaceWarpExtensionWrapper() {
+	singleton = nullptr;
+}
+
+HashMap<String, bool *> OpenXRFbSpaceWarpExtensionWrapper::get_requested_extensions() {
+	HashMap<String, bool *> request_extensions;
+
+	request_extensions[XR_FB_SPACE_WARP_EXTENSION_NAME] = &available;
+
+	return request_extensions;
+}
+
+void *OpenXRFbSpaceWarpExtensionWrapper::set_system_properties_and_get_next_pointer(void *p_next_pointer) {
+	return &system_space_warp_properties;
+}
```

Comment:
This needs to set the `p_next_pointer`, otherwise, we could lose some of the structs in the next pointer chain.

Something like:

```suggestion
void *OpenXRFbSpaceWarpExtensionWrapper::set_system_properties_and_get_next_pointer(void *p_next_pointer) {
	system_space_warp_properties.next_pointer = p_next_pointer;
	return &system_space_warp_properties;
}
```